src/data/local_storage.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * @module Data
+ * @submodule LocalStorage
+ * @requires core
+ *
+ * This module defines the p5 methods for working with local storage
+ */
+
+'use strict';
+var p5 = require('../core/main');
+/**
+ *
+ * Stores a value in local storage under the key name. The key can
+ * be the name of the variable but it does not have to be. To retrieve
+ * stored items see <a href="#/p5/loadItem">loadItem</a>.
+ *
+ * @method storeItem
+ * @for p5
+ * @param {String} key
+ * @param {String|Number|Object|Boolean|p5.Color} value
+ *
+ * @example
+ * <div><code>
+ * // Type to change the letter in the
+ * // center of the canvas.
+ * // If you reload the page, it will
+ * // still display the last key you entered
+ *
+ * let myText;
+ *
+ * function setup() {
+ *   createCanvas(100, 100);
+ *   myText = loadItem('myText');
+ *   if (myText === null) {
+ *     myText = '';
+ *   }
+ * }
+ *
+ * function draw() {
+ *   textSize(40);
+ *   background(255);
+ *   text(myText, width / 2, height / 2);
+ * }
+ *
+ * function keyPressed() {
+ *   myText = key;
+ *   storeItem('myText', myText);
+ * }
+ * </code></div>
+ *
+ * @alt
+ * When you type the key name is displayed as black text on white background.
+ * If you reload the page, the last letter typed is still displaying.
+ *
+ */
+p5.prototype.storeItem = function(key, value) {
+  if (value === 'undefined') {
+    console.log('You cannot store undefined variables using storeItem()');
+  }
+  var type = typeof value;
+  switch (type) {
+    case 'number':
+    case 'boolean':
+      value = value.toString();
+      break;
+    case 'object':
+      value = JSON.stringify(value);
+      break;
+    case 'string':
+    default:
+      break;
+  }
+
+  localStorage.setItem(key, value);
+  var typeKey = key + 'Type';
```

Comment:
I would suggest making this a little more likely to be unique, so there's no chance of it being accidentally overwritten by someone calling `storeItem()` with "Type" as part of the key name. I know this isn't super likely, but it's possible.

Maybe its just a matter of modifying this to "[Type]" or something similar.

Reply:
i wonder if we should prefix all items and typeItems stored by p5 with something like `[p5-ls]` so it's harder to get accidental collisions too? that might be overkill though.